JenkinsOKEfile

```diff
@@ -0,0 +1,594 @@
+                    }
+                    steps {
+                        sh 'echo ${DOCKERHUB_PASSWORD} | docker login -u ${DOCKERHUB_USERNAME} --password-stdin'
+                        sh "mvn -DtrimStackTrace=false clean install"
+                    }
+                }
+
+                stage('Run Sonar Analysis') {
+                    steps {
+                        sh '''
+                            rm -rf ${WORKSPACE}/.mvn/maven.config
+                            mkdir -p ${WORKSPACE}/.mvn
+                            touch ${WORKSPACE}/.mvn/maven.config
+                            echo "-Dsonar.projectKey=${sonar_project_key}"                        >> ${WORKSPACE}/.mvn/maven.config
+                            if [ -z "${CHANGE_ID}" ]; then
+                                echo "-Dsonar.branch.name=${BRANCH_NAME}"                         >> ${WORKSPACE}/.mvn/maven.config
+                            else
+                                echo "-Dsonar.pullrequest.provider=GitHub"                        >> ${WORKSPACE}/.mvn/maven.config
+                                echo "-Dsonar.pullrequest.github.repository=${sonar_github_repo}" >> ${WORKSPACE}/.mvn/maven.config
+                                echo "-Dsonar.pullrequest.key=${CHANGE_ID}"                       >> ${WORKSPACE}/.mvn/maven.config
+                                echo "-Dsonar.pullrequest.branch=${CHANGE_BRANCH}"                >> ${WORKSPACE}/.mvn/maven.config
+                                echo "-Dsonar.pullrequest.base=${CHANGE_TARGET}"                  >> ${WORKSPACE}/.mvn/maven.config
+                            fi
+                            echo "${WORKSPACE}/.mvn/maven.config contents:"
+                            cat "${WORKSPACE}/.mvn/maven.config"
+                        '''
+                        withSonarQubeEnv('SonarCloud') {
+                            sh "mvn sonar:sonar"
+                        }
+                    }
+                }
+
+                 stage('Verify Sonar Quality Gate') {
+                    steps {
+                        timeout(time: 10, unit: 'MINUTES') {
+                            // Set abortPipeline to true to stop the build if the Quality Gate is not met.
+                            waitForQualityGate(abortPipeline: false, webhookSecretId: "${sonar_webhook_secret_creds}")
+                        }
+                    }
+                 }
+
+                stage('Make Workspace bin directory') {
+                    steps {
+                        sh "mkdir -m777 -p ${WORKSPACE}/bin"
+                    }
+                }
+                stage ('Install oci cli') {
+                    environment {
+                        runtime_path = "${WORKSPACE}/bin:${PATH}:/bin"
+                    }
+                    steps {
+                        sh '''
+                            export PATH=${runtime_path}
```

Comment:
Installing/configuring software at a global level is a really unfriendly thing to do in a shared environment.   You should consider another way to handle this.

Reply:
fixed all comments, waiting to get python3 to be part of the oci image creation. Meanwhile everything else is ready for review. Moved to single branch pipeline : https://build.weblogick8s.org:8443/job/wko-oke-test-seqential/11/